tree/ntuple/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -65,6 +67,130 @@ public:
+   void BindValue(ROOT::RFieldToken token, std::shared_ptr<void> objPtr) { fEntry->BindValue(token, objPtr); }
+
+   template <typename T>
+   std::shared_ptr<T> GetPtr(std::string_view fieldName) const
+   {
+      if (fIsValid)
+         return fEntry->GetPtr<T>(fieldName);
+
+      return nullptr;
+   }
+
+   template <typename T>
+   std::shared_ptr<T> GetPtr(ROOT::RFieldToken token) const
+   {
+      if (fIsValid)
+         return fEntry->GetPtr<T>(token);
+
+      return nullptr;
+   }
+
+   void Read(ROOT::NTupleSize_t index)
+   {
+      assert(fIsValid && "cannot read invalidated entries");
+      fEntry->Read(index);
+   }
+
+   const std::string &FindFieldName(ROOT::RFieldToken token) const { return fEntry->FindFieldName(token); }
+
+   static std::unique_ptr<RNTupleProcessorEntry> Create(std::unique_ptr<ROOT::REntry> entry)
+   {
+      return std::unique_ptr<RNTupleProcessorEntry>(new RNTupleProcessorEntry(std::move(entry)));
+   }
+
+public:
+   RNTupleProcessorEntry() = delete;
+   RNTupleProcessorEntry(const RNTupleProcessorEntry &other) = delete;
+   RNTupleProcessorEntry &operator=(const RNTupleProcessorEntry &other) = delete;
+   RNTupleProcessorEntry(RNTupleProcessorEntry &&other) = default;
+   RNTupleProcessorEntry &operator=(RNTupleProcessorEntry &&other) = default;
+   ~RNTupleProcessorEntry() = default;
+
+   bool IsValid() const { return fIsValid; }
+
+   ROOT::RFieldToken GetToken(std::string_view fieldName) const { return fEntry->GetToken(fieldName); }
+
+   ROOT::REntry::ConstIterator_t begin() const { return fEntry->begin(); }
+   ROOT::REntry::ConstIterator_t end() const { return fEntry->end(); }
+};
+
+template <typename T>
+class RNTupleProcessorValuePtr {
+   friend class RNTupleProcessor;
+   friend class RNTupleProcessorEntry;
+
+private:
+   const RNTupleProcessorEntry *fProcessorEntry;
+   ROOT::RFieldToken fToken;
+
+   RNTupleProcessorValuePtr(std::string_view fieldName, const RNTupleProcessorEntry *processorEntry)
+      : fProcessorEntry(processorEntry), fToken(processorEntry->GetToken(fieldName))
+   {
+   }
+
+public:
+   bool HasValue() { return fProcessorEntry->IsValid(); }
+
+   std::shared_ptr<T> GetPtr() const
+   {
+      if (fProcessorEntry->IsValid())
+         return fProcessorEntry->GetPtr<T>(fToken);
+
+      return nullptr;
+   }
+
+   const T &operator*()
+   {
+      if (auto ptr = GetPtr(); ptr)
+         return *std::static_pointer_cast<T>(ptr);
+      else
+         throw RException(R__FAIL("cannot read \"" + fProcessorEntry->FindFieldName(fToken) +
+                                  "\" because the entry it belongs to is invalid"));
+   }
+};
+
+template <>
+class RNTupleProcessorValuePtr<void> {
+   friend class RNTupleProcessor;
+
+private:
+   const RNTupleProcessorEntry *fProcessorEntry;
+   ROOT::RFieldToken fToken;
+
+   RNTupleProcessorValuePtr(std::string_view fieldName, const RNTupleProcessorEntry *processorEntry)
```

Comment:
Can `processorEntry` be null here? If not, I'd pass it by reference (but it's fine to store it by pointer ofc)